### PR TITLE
Stamina changes [TMC]

### PR DESCRIPTION
**Pull request by 4dplanner**
## About The Pull Request
Stamina damage no longer stacks with regular damage.
Stamina regen is paused for 10 seconds every time you take stamina damage
Stamina damage is purged if stamina regen resumes after you enter stamcrit

changes to make this work:
disablers knocked down to 30 
punch damage changed to full raw, 1.5x stamina.
flesh-eating symptom deals 2x stamina damage
CQC unchanged
Bostaff manually stacks
Energised jelly blob deals full stamina damage
Pressurised slime blob deals full stamina damage
Blood boil deals 1.5x stamina damage to servants of ratvar (why do I bother?)
Revenant blight now deals 20 stamina damage
Space carp no longer deal stamina damage, but deal 20 brute.
Enchanted bullets deal 20 brute damage
Beanbags reduced to 55 stamina damage 
Rubbershot damage untouched
Ebow stamina damage increased to 60

bone hurting juice deals 7.5 stamina damage per tick, down from 15

## Why It's Good For The Game
Stamina damage not stacking with regular damage is meant to make disablers less good as a weapon for all-out firefights, but remain good for arresting criminals as a sec squad.

The stamina regen pause allows for chemicals that deal stamina damage to be effective even if they don't only target one body part (as in #43924), and gets rid of the counterintuitive speedy regeneration if multiple bodyparts are damaged. It also makes stamina damage calculations more straightforward - currently 4 damage/tick to the chest is really 1, and adding 25% more damage to the chem deals 100% more damage per tick.

The stamina purge is to make stamcrit less of a confirmed win - several times as sec officer I've succumbed to poison, run out of charge, etc only for the rest of sec to arrive before the perp awakes from his 3 hour sleep. Stamina damage is fully purged to make keeping someone down require more attention, since under the current system even if they somehow get up they'll be so slow it'll be trivial to put them down again. Basically, makes handcuffs more useful again.

## Changelog
:cl:
add: stamina damage now pauses natural stamina regen for 10 seconds
add: stamina regenerates fully when it can.
balance: disabler damage reduced 
tweak: stamina damage no longer stacks with normal damage for the purposes of damage slowdown and crit
balance: many sources of stamina damage have been adjusted, check PR for details
/:cl:

### CODE REVIEW

**Comment by 4dplanner** [comment]
@PKPenguin321

**Comment by actioninja** [comment]
Big, bold, sensual.
This might interact poorly with that stam chem rework if that gets merged

**Comment by 4dplanner** [comment]
Shouldn't cause too much of a problem - that rework makes stamina damage all deal to the chest, so regen didn't have as much of an effect anyway. It is still a buff to stamina damage chems, however.

**Comment by nemvar** [comment]
Stamina reagents dealing damage to every body part is still stupid and if this gets merged will result in people suddenly losing control of all their limbs at the same time.

**Comment by 4dplanner** [comment]
Crit threshold should be less than the amount required to disable limbs IIRC. I was more thinking of this as potential for future chem effects, such as a chem which deals damage to both legs.

**Comment by 4dplanner** [comment]
Just realised damaging both legs is what your tirizene does. I think that pausing regen helps make the damage calculations more straightforward there, rather than having to look at how many bodyparts are being hit to calculate DPS.

**Comment by Tlaltecuhtli** [comment]
>Stamina regen is paused for 10 seconds every time you take stamina damage

does it only stop only natural stamina regen or even chem stamina regen?

**Comment by 4dplanner** [comment]
> > Stamina regen is paused for 10 seconds every time you take stamina damage
> 
> does it only stop only natural stamina regen or even chem stamina regen?

just natural stamina regen

**Comment by HomocidlePancakes** [comment]
Good PR

**Comment by Shadowflame909** [comment]
Weren't disablers suffering from an unfixed instant slowdown bug?

**Comment by 4dplanner** [comment]
> Weren't disablers suffering from an unfixed instant slowdown bug?

I don't know what you're referring to, but disablers were meant to instantly slow down. They deal 40 damage, which is the damage slowdown threshold.

**Comment by HomocidlePancakes** [comment]
They are supposed to deal 37. Nani?

**Comment by 4dplanner** [comment]
damage
@HomocidlePancakes

**Comment by 4dplanner** [comment]
See #41169

**Comment by Shadowflame909** [comment]
Was told it was due to a glitch with limb targeting. What crazy coder gives a gun nearly 40 shots and has it where if you get hit by one, it's a win?

**Comment by necromanceranne** [comment]
Another buff to unarmed combat. Outrageous! Can the coders be stopped before every assistant is a super saiyan waiting to be unleashed.

**Comment by Jimmius** [comment]
good change

**Comment by OverlyGeneric** [comment]
Take a shot for every time someone says stamina after this post.

**Comment by 4dplanner** [comment]
To clarify, the punch miss chance interaction isn't affected.

**Comment by PKPenguin321** [comment]
Love it! Only one thing I would add: 
>Stamina regen is paused for 10 seconds every time you take stamina damage

This might work better if the pause time scaled to stamina damage taken. A disabler shot could pause for a full 10 seconds, but a 5-damage punch should only pause for maybe half or even a third of that. I don't have an exact formula in mind but it doesn't need to be too exact.

**Comment by ATH1909** [comment]
I like this PR... except for the "all stamina damage is purged when you leave staminacrit" part, which I think makes little sense and might lead to dumb tactics like you intentionally punching yourself into stamina crit in order to take only about 10 seconds to remove all of your stamina damage instead of however long it'd normally take to heal off your stamina damage with natural regen.

Maybe just increase stamina regen while in staminacrit, but make it so that you need to go below 60 points of stamina damage (in other words, go above 40 points of positive stamina health) in order to leave stamina crit?

I'm also not particularly a fan of making stamina damage not stack with regular damage, which I don't think really makes sense.

Actually, I think I'm just not a fan of this PR in general (except for the "Stamina regen is paused for 10 seconds every time you take stamina damage" part, which I _do_ like and think should be made a thing in the game).

**Comment by zxaber** [comment]
Do we really need to nerf disabler damage? Why are we so married to nerfing security?

The rest of the PR makes sense and I like.

**Comment by necromanceranne** [comment]
What this does do is make mixed damage sources (besides unarmed due to how unarmed mechanics work) a bit weaker. Rubbershot I feel might lose out on this one.

**Comment by YakumoChen** [comment]
There's a lot of sources of stamina damage that are also pretty reliant on mixed damage. Some of the blob types are major ones.

**Comment by Shadowflame909** [comment]
I guess I should hold off on giving the bananium e-sword stamina damage to make it useful again.

**Comment by Tlaltecuhtli** [comment]
it seems weird that you have to shoot multiple times someone 1 punch away from crit, it also affects the pkp rework which makes punches give stamina + dmg, i would say maybe have the stamina dmg still dependent to the health

**Comment by 4dplanner** [comment]
As to mixed damage balance issues: yes, this PR does need a balance pass. I should probably tag it as WIP.

As to whether the change to mixed damage should be there at all, I think it should. It makes lethal and nonlethal methods of arrest more distinct - lethal damage builds up, while nonlethal arrests are always just as hard each time but have high damage and lots of ammo.

>Do we really need to nerf disabler damage? Why are we so married to nerfing security?

With stamina regen blocked, a single disabler shot on an unarmored target would be a slowdown for ten seconds, which would be crazy. Disablers will still be very strong (1.5x laser damage), and the change to stamina regen will also make them more effective if you randomly hit an arm once instead of the chest.

>This might work better if the pause time scaled to stamina damage taken.

I'm not sure about this - it's usually similarly easy to disengage regardless of the damage of the strike, and I'd rather not have it be too easy to stall low-damage (such as punch) fights by disengaging for a very short time.

**Comment by Fox-McCloud** [comment]
last remaining toxic thing about stamina, if this goes through, would be the slow.

Once that's gone, stamina will probably be in a good place.

**Comment by 4dplanner** [comment]
I don't think removing the slow would be a good idea.

**Comment by Shadowflame909** [comment]
We need to nerf the ammo, not the slow.

40 bullets but if one hits I lose? Great balance security

**Comment by anconfuzedrock** [comment]
am I misreading this, or does this mean punches do overall 2.5x damage, making them better than most objects for combat? or is this saying punches do zero damage? I'm not sure which idea I dislike more.

**Comment by 4dplanner** [comment]
> am I misreading this, or does this mean punches do overall 2.5x damage, making them better than most objects for combat? or is this saying punches do zero damage? I'm not sure which idea I dislike more.

They do 2.5x damage, but since stamina and brute damage don't stack it's effectively the same as the current system.

**Comment by Fox-McCloud** [comment]
>I don't think removing the slow would be a good idea.

It alone is pretty much responsible for stamina having little to no counterplay around it.

I think there's a case to be made for disabler beams, themselves, to impart a flat slow (that doesn't stack, and only last for a short duration), but the system of "get his by 1 disabler beam and be slowed forever" is what makes disabler using fairly toxic.

**Comment by 4dplanner** [comment]
To be clear, after this PR the slow will only kick in after 2 hits, not 1.

**Comment by Fox-McCloud** [comment]
Not sure that's going to make much of a differences, really, to be honest.

File: code/game/objects/items/defib.dm [hunk 1]
```diff
@@ -467,7 +467,7 @@
 			"<span class='userdanger'>[user] has touched [M] with [src]!</span>")
 	M.adjustStaminaLoss(50)
 	M.Paralyze(100)
-	M.updatehealth() //forces health update before next life tick
+	M.updatehealth() //forces health update before next life tick //isn't this done by adjustStaminaLoss anyway?
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/defib.dm]: only you can read code

**Comment by Tlaltecuhtli** [comment]
its dumb that only punches deal dmg + stamina, it should be every object, 
like in wwe wrestling you can see ppl staminacrit’d by punches but also by folded chairs

**Comment by 4dplanner** [comment]
I am thinking it might be more consistent/generally better to have stamina regen ALWAYS be full, whether you're stunned or not. This removes the counterintuitive aspect, and also gets rid of situations where it's better to not stun someone to leave them slowed forever.

**Comment by YakumoChen** [comment]
The slow in stamina is the only thing that makes using the disabler bearable, since hitting someone with 3-4 beams in a reasonably short time would be much, much more difficult if they were full speed yakety sax the whole time. At that point just relying on lethals for slows would be a lot more lucrative if it becomes that hard to disable anyone.

**Comment by PKPenguin321** [comment]
😤🙏

**Pull request merged**